### ISSUE TITLE
Next and Previous buttons not working correctly

### ISSUE THREAD

**Issue by bcarroll** [generaldave/simpleMP3player#3]
The Next and Previous buttons are not moving through the list of songs sequentially.

Seems more like Previous = random backwards and Next = random forwards through the list of songs.
-----------------------------------------------------------------------------------------------------------------
Windows 10
Python = 3.6.2
pygame = 1.9.3

**Comment by generaldave** [comment]
Yeah, I noticed there was an issue with this app a while ago. It worked perfectly fine for several months and one day it all of a sudden did not, even though I had not made any changes. Maybe it has to do with Python updates. I don't really know. I've been wanting to get back to this app and fix things, but honestly, I would like to just rewrite the whole thing being the structuring is terrible. The problem is that I just don't have the time right now between school, work, and freelance type work.

Either way, thank you for submitting the issue and for fixing some of the comments to accurately reflect the blocks and classes they were referencing.

~Dave

**Comment by bcarroll** [comment]
I looked theough the code and didn’t see anything that looked wrong with the logic.

P.S.  the interface looks great!

**Comment by bcarroll** [comment]
Accidently closed this issue...

Not sure if this is the root of the issue or not, but SongsBlock.py needs refactoring/debugging.

I added a debug logging statement to every method in SongBlock.py and noticed the redundant method calls with every iteration of the pygame loop (they are redundant because the information doesn't change while a song is playing):

songcount = self.songs.getSongCount()
...
previousSong  = self.songs.getTitle(index)[:32]
...
nextSong  = self.songs.getTitle(index)[:32]

**Comment by generaldave** [comment]
When/if I get back to it, I'll make sure to pay attention to redundant calls...which makes me realize there are redundant calls in pretty much all of my Pygame applications because they all call updates in every loop even though information does not change. Thank you and thank you for the complement on the interface.

**Comment by generaldave** [comment]
So, I did some researching on this. It seems that not all mp3 files are compatible with Pygame. This is most likely the issue with skipping some songs. I could get around this by using vlc.py (https://github.com/oaubert/python-vlc), but this requires the user to have VLC Media Player installed on their machine. The other option is to convert non-playable mp3 files to ogg, so that they are compatible with Pygame. I think the second option is more trouble than it's worth. As I would think most people would have VLC installed on their machines, I am leaning toward integrating vlc.py.

**Comment by bcarroll** [comment]
Perhaps [PySoundFile ](https://pypi.python.org/pypi/PySoundFile/0.8.1) would be a better approach as it doesn't appear to require any external binaries.

**Comment by CodeJish** [comment]
hey when will this be fixed

**Comment by generaldave** [comment]
CodeJish, assuming the problem is what I thought above, updating Pygame solved the issue for me. I tested the first 100,000,000 "frames", which is like 4 - 5 seconds, of the 527 mp3s I have on the newest version (2.1.2) of Pygame. All of them played with no problem. However, on 1.9.3, they wouldn't all play.

Hope that helps.

Let me know if there is still an issue.
